### ISSUE TITLE
no-undefined-types: false positive for @template on exported function

### ISSUE THREAD

**Issue by kevinoid** [gajus/eslint-plugin-jsdoc#578]
## Expected behavior

The code below would not produce any warnings.

## Actual behavior

    /tmp/example.js
      4:0  warning  The type 'ValueType' is undefined  jsdoc/no-undefined-types
      5:0  warning  The type 'ValueType' is undefined  jsdoc/no-undefined-types
    
    ✖ 2 problems (0 errors, 2 warnings)

Note that if `exports.resolve1 = ` is removed, the warnings do not occur.

## ESLint Config

```json
{
  "env": {
    "es6": true
  },
  "extends": [
    "plugin:jsdoc/recommended"
  ],
  "settings": {
    "jsdoc": {
      "mode": "typescript"
    }
  }
}
```

## ESLint sample

```js
/** Gets a Promise resolved with a given value.
 *
 * @template ValueType
 * @param {ValueType} value Value to resolve.
 * @returns {Promise<ValueType>} Promise resolved with value.
 */
exports.resolve1 = function resolve1(value) {
  return Promise.resolve(value);
};
```

## Environment

* Node version: v14.0.0
* ESLint version v7.2.0
* `eslint-plugin-jsdoc` version: 27.0.7

## Related Issues

The issue may be related to #559, but since it occurs on the function where `@template` is used rather than inside, I thought it may be a separate issue.

**Comment by kevinoid** [comment]
I can confirm 27.1.1 fixes the issue for me.  Thanks for the fast fix @brettz9 and @gajus!
